src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
+            IValueBinder binder = (IValueBinder)Activator.CreateInstance(genericType, context);
+
+            return Task.FromResult(binder);
+        }
+
+        internal IMobileServiceClient BindForClient(MobileTableAttribute attribute)
+        {
+            MobileTableContext context = CreateContext(attribute);
+            return context.Client;
+        }
+
+        internal MobileTableContext BindForOutput(MobileTableAttribute attribute)
+        {
+            return CreateContext(attribute);
+        }
+
+        internal static Uri ResolveMobileAppUri(Uri defaultUri, string attributeUriString)
+        {
+            Uri resolvedUri = defaultUri;
+
+            if (!string.IsNullOrEmpty(attributeUriString))
+            {
+                string resolvedUriString = GetSettingFromConfigOrEnvironment(attributeUriString);
+                resolvedUri = new Uri(resolvedUriString);
+            }
+
+            return resolvedUri;
+        }
+
+        internal MobileTableContext CreateContext(MobileTableAttribute attribute)
+        {
+            Uri resolvedUri = ResolveMobileAppUri(MobileAppUri, attribute.MobileAppUri);
+            string resolvedApiKey = ResolveApiKey(ApiKey, attribute.ApiKey);
+
+            return new MobileTableContext
+            {
+                Client = GetClient(resolvedUri, resolvedApiKey),
+                ResolvedAttribute = attribute
+            };
+        }
+
+        internal static string ResolveApiKey(string defaultApiKey, string attributeApiKey)
+        {
+            string resolvedApiKey = defaultApiKey;
+
+            // If the attribute specifies an empty string ApiKey, set the ApiKey to null.
+            if (attributeApiKey == string.Empty)
+            {
+                resolvedApiKey = null;
+            }
+            else if (attributeApiKey != null)
+            {
+                resolvedApiKey = GetSettingFromConfigOrEnvironment(attributeApiKey);
+            }
+
+            return resolvedApiKey;
+        }
+
+        internal IMobileServiceClient GetClient(Uri mobileAppUri, string apiKey)
+        {
+            string key = GetCacheKey(mobileAppUri, apiKey);
+            return ClientCache.GetOrAdd(key, (c) => CreateMobileServiceClient(ClientFactory, mobileAppUri, apiKey));
+        }
+
+        internal static string GetCacheKey(Uri mobileAppUri, string apiKey)
+        {
+            return string.Format("{0};{1}", mobileAppUri, apiKey);
+        }
+
+        internal static IMobileServiceClient CreateMobileServiceClient(IMobileServiceClientFactory factory, Uri mobileAppUri, string apiKey = null)
+        {
+            HttpMessageHandler[] handlers = null;
+            if (!string.IsNullOrEmpty(apiKey))
+            {
+                handlers = new[] { new MobileServiceApiKeyHandler(apiKey) };
+            }
+
+            return factory.CreateClient(mobileAppUri, handlers);
         }
 
         internal static string GetSettingFromConfigOrEnvironment(string key)
```

Comment:
You could have the ctor require a INameResolver;  or we could leverage AttributeCloner. 
I believe in ScriptHost (Functions), the configuration is supplied with a NameResolver that does this.  
[https://github.com/Azure/azure-webjobs-sdk-script/blob/21f1cd77957ec9107e68e938ece8af1886becc92/src/WebJobs.Script/Config/NameResolver.cs](https://github.com/Azure/azure-webjobs-sdk-script/blob/21f1cd77957ec9107e68e938ece8af1886becc92/src/WebJobs.Script/Config/NameResolver.cs) 

---
In reply to: [63604093](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/97#discussion_r63604093) [](ancestors = 63604093,63600341)